openmessaging/src/main/java/io/openmessaging/rocketmq/consumer/PullConsumerImpl.java

```diff
@@ -76,110 +93,268 @@ public PullConsumerImpl(final KeyValue properties) {
+        for (String queueName : queueNames) {
+            bindQueue(queueName);
+        }
+    }
+
+    @Override
+    public void bindQueue(String queueName, MessageListener listener) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void bindQueues(List<String> queueNames, MessageListener listener) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void bindQueue(String queueName, BatchMessageListener listener) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void bindQueues(List<String> queueNames, BatchMessageListener listener) {
+        throw new UnsupportedOperationException();
+    }
+
+    @Override
+    public void unbindQueue(String queueName) {
         this.rocketmqPullConsumer.getRegisterTopics().remove(queueName);
-        return this;
     }
 
     @Override
-    public Message receive() {
-        MessageExt rmqMsg = localMessageCache.poll();
+    public void unbindQueues(List<String> queueNames) {
+        for (String queueName : queueNames) {
+            this.rocketmqPullConsumer.getRegisterTopics().remove(queueName);
+        }
+    }
+
+    @Override
+    public boolean isBindQueue() {
+        Set<String> registerTopics = rocketmqPullConsumer.getRegisterTopics();
+        if (null == registerTopics || registerTopics.isEmpty()) {
+            return false;
+        }
+        return true;
+    }
+
+    @Override
+    public List<String> getBindQueues() {
+        Set<String> registerTopics = rocketmqPullConsumer.getRegisterTopics();
+        return new ArrayList<>(registerTopics);
+    }
+
+    @Override
+    public void addInterceptor(ConsumerInterceptor interceptor) {
+        consumerInterceptors.add(interceptor);
+    }
+
+    @Override
+    public void removeInterceptor(ConsumerInterceptor interceptor) {
+        consumerInterceptors.remove(interceptor);
+    }
+
+    @Override
+    public Message receive(long timeout) {
+        KeyValue properties = new DefaultKeyValue();
+        properties.put("TIMEOUT", timeout);
+        MessageExt rmqMsg = localMessageCache.poll(properties);
         return rmqMsg == null ? null : OMSUtil.msgConvert(rmqMsg);
     }
 
     @Override
-    public Message receive(final KeyValue properties) {
+    public Message receive(String queueName, int partitionId, long receiptId, long timeout) {
+        KeyValue properties = new DefaultKeyValue();
+        properties.put("QUEUE_NAME", queueName);
+        properties.put("PARTITION_ID", partitionId);
+        properties.put("RECEIPT_ID", receiptId);
+        properties.put("TIMEOUT", timeout);
         MessageExt rmqMsg = localMessageCache.poll(properties);
         return rmqMsg == null ? null : OMSUtil.msgConvert(rmqMsg);
     }
 
     @Override
-    public void ack(final String messageId) {
-        localMessageCache.ack(messageId);
+    public List<Message> batchReceive(long timeout) {
```

Comment:
In the batchReceive and localMessageCache.batchPoll  method,the timeout parameter won't have any effect.

Reply:
It has been modified